lib/util/data_view_reader.js

```diff
@@ -174,6 +174,42 @@ shaka.util.DataViewReader = class {
   }
 
 
+  /**
+   * Reads a 64 bit integer, and advances the reader.
+   * @return {bigint} The integer.
+   * @export
+   */
+  readBigInt64() {
+    /** @type {number} */
+    let low;
+    /** @type {number} */
+    let high;
+
+    try {
+      if (this.littleEndian_) {
+        low = this.dataView_.getUint32(this.position_, true);
+        high = this.dataView_.getUint32(this.position_ + 4, true);
+      } else {
+        high = this.dataView_.getUint32(this.position_, false);
+        low = this.dataView_.getUint32(this.position_ + 4, false);
+      }
+    } catch (exception) {
+      throw this.outOfBounds_();
+    }
+
+    if (!window.BigInt) {
+      throw new shaka.util.Error(
+          shaka.util.Error.Severity.CRITICAL,
+          shaka.util.Error.Category.MEDIA,
+          shaka.util.Error.Code.JS_INTEGER_OVERFLOW);
+    }
+
+    this.position_ += 8;
+
+    return BigInt((high * Math.pow(2, 32)) + low);
```

Comment:
This seems to imply that Number can be up-cast to BigInt automatically in the multiplication of high * Math.pow(2, 32).  If not, there is already precision lost here, isn't there?

Reply:
It is converted automatically, so there is no loss of precision.

Reply:
I believe you are mistaken.  Run this is the JS debugger, and you will see:

```js
> high = 8675309
> low = 4390116

> BigInt((high * Math.pow(2, 32)) + low) == (BigInt(high) * BigInt(Math.pow(2, 32))) + BigInt(low)
false
```

Your expression is the one on the left.  It does `high * pow + low`, all with Numbers, then converts the already-rounded-off result to BigInt.  If instead you convert each piece into BigInt first, you get a precise int that differs from the one you're computing.

Reply:
I changed the implementation